src/app/portlets/dot-edit-content/components/dot-edit-content-view-as-toolbar/dot-edit-content-view-as-toolbar.component.spec.ts

```diff
@@ -0,0 +1,133 @@
+            languageId: 1,
+            lastReview: '2016-02-19 11:21:03.788',
+            modDate: '2016-02-19 11:21:03.802',
+            modUser: 'dotcms.org.1',
+            name: 'Manager',
+            owner: 'dotcms.org.1',
+            photoContentAsset: '1c56ba62-1f41-4b81-bd62-b6eacff3ad23/photo',
+            sortOrder: 0,
+            stInode: 'c938b15f-bcb6-49ef-8651-14d455a97045'
+        },
+        language: { id: 1, languageCode: 'en', countryCode: 'US', language: 'English', country: 'United States' },
+        device: { id: '1', label: 'iPhone', width: '375px', height: '667px' }
+    };
+    const pageViewAsInitialValue: DotEditPageViewAs = {
+        persona: { identifier: '0' },
+        language: { id: 1, languageCode: 'en', countryCode: 'US', language: 'English', country: 'United States' },
+        device: { id: '0', label: 'Desktop', width: '100%', height: '100%' }
+    };
+
+    beforeEach(() => {
+        const messageServiceMock = new MockDotMessageService({
+            'modes.persona.no.persona': 'Default Persona'
+        });
+
+        DOTTestBed.configureTestingModule({
+            declarations: [DotEditContentViewAsToolbarComponent],
+            imports: [BrowserAnimationsModule],
+            providers: [
+                DotViewAsService,
+                {
+                    provide: DotDevicesService,
+                    useClass: DotDevicesServiceMock
+                },
+                {
+                    provide: DotPersonasService,
+                    useClass: DotPersonasServiceMock
+                },
+                {
+                    provide: DotLanguagesService,
+                    useClass: DotLanguagesServiceMock
+                },
+                {
+                    provide: DotMessageService,
+                    useValue: messageServiceMock
+                }
+            ]
+        });
+
+        fixture = DOTTestBed.createComponent(DotEditContentViewAsToolbarComponent);
+        component = fixture.componentInstance;
+        de = fixture.debugElement;
+        dotViewAsService = fixture.debugElement.injector.get(DotViewAsService);
+    });
+
+    it('should set the initial values for the dropdowns & emit the Device', () => {
+        spyOn(component.changeDevice, 'emit');
+        fixture.detectChanges();
+        expect(component.viewAsConfig).toEqual(pageViewAsInitialValue);
+        expect(component.changeDevice.emit).toHaveBeenCalledWith(pageViewAsInitialValue.device);
+    });
+
+    it('should load dropdowns with DotViewAsService values', () => {
+        dotViewAsService.selected = pageViewAsPreLoadedValue;
+        fixture.detectChanges();
+
+        expect(component.viewAsConfig).toEqual(pageViewAsPreLoadedValue);
+    });
+
+    it('should emit changes in Language & save the new values in DotViewAsService', () => {
+        const pDropDown: DebugElement = de.query(By.css('.view-as-toolbar-languages'));
+        component.ngOnInit();
```

Comment:
Why you didn't use ```fixture.detectChanges()``` instead?

Reply:
We dont should call ngOnInit directly into the test